src/citrine/ara/variables.py

```diff
@@ -37,12 +37,9 @@ def _attrs(self) -> List[str]:
         pass  # pragma: no cover
 
     def __eq__(self, other):
-        try:
-            return all([
-                self.__getattribute__(key) == other.__getattribute__(key) for key in self._attrs()
-            ])
-        except AttributeError:
-            return False
+        return type(self) == type(other) and all([
```

Comment:
Do we want strict type equality?  What if we subclass?

Reply:
Would we want to different child types to be able to be equal?

Reply:
There are variables with the exact same data other than type hint right now. The type is definitely a distinguishing factor right now.

Reply:
Sorry, that isn't accurate. I have been documenting the *column* definitions recently and it is true of them. For variables, there is no possible for exact JSON equality when you remove the type hint as of right now, but I wouldn't be surprised if we did run into that in the future since there is a lot of attribute overlap, and these `Variable` classes really model functions rather than data.